### PR TITLE
Fix to infinite loop with listviewHeaders

### PR DESCRIPTION
**Pull request by NicolasChampagne**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_workitems/edit/158362

### CODE REVIEW

**Comment by ghuntley** [comment]
Thanks for adding the labels :)

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs [hunk 1]
```diff
@@ -26,5 +26,19 @@ public void ListView_ListViewVariableItemHeightLong_InitializesTest()
 			// Assert initial state
 			Assert.IsNotNull(theListView.GetDependencyPropertyValue("DataContext"));
 		}
+
+		[Test]
+		[AutoRetry]
+		[ActivePlatforms(Platform.iOS, Platform.Android)]
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: Could you indicate where the test fails on wasm?

**Comment by jcantin-nventive** [comment]
This PR also fixes the following internal issue, which has to do with group headers
https://nventive.visualstudio.com/Umbrella/_workitems/edit/158452

**Pull request merged**